mlkem/reduce.c

```diff
@@ -3,6 +3,29 @@
 #include <stdint.h>
 #include "params.h"
 
+
+/*************************************************
+ * Name:        cast_uint16_to_int16
+ *
+ * Description: Cast uint16 value to int16
+ *
+ * Returns:
+ *   input x in     0 .. 32767: returns value unchanged
+ *   input x in 32768 .. 65535: returns (x - 65536)
+ **************************************************/
+static inline int16_t cast_uint16_to_int16(uint16_t x) {
+  // PORTABILITY: This relies on uint16_t -> int16_t
+  // being implemented as the inverse of int16_t -> uint16_t,
+  // which is implementation-defined (C99 6.3.1.3 (3))
+  //
+  // CBMC (correctly) fails to prove this conversion is OK,
+  // so we have to suppress that check here
+#pragma CPROVER check push
```

Comment:
Can you hoist those `#pragma` out of the function and bracket them with `#if defined(CBMC)`? This way, we hopefully don't get issues with compilation in non-CBMC mode because of unknown pragmas.

Reply:
Good idea